packages/twenty-server/src/modules/invitation/standard-objects/invitation.workspace-entity.ts

```diff
@@ -0,0 +1,53 @@
+import { FieldMetadataType } from 'src/engine/metadata-modules/field-metadata/field-metadata.entity';
+import { BaseWorkspaceEntity } from 'src/engine/twenty-orm/base.workspace-entity';
+import { WorkspaceEntity } from 'src/engine/twenty-orm/decorators/workspace-entity.decorator';
+import { WorkspaceField } from 'src/engine/twenty-orm/decorators/workspace-field.decorator';
+import { INVITATION_STANDARD_FIELD_IDS } from 'src/engine/workspace-manager/workspace-sync-metadata/constants/standard-field-ids';
+import { STANDARD_OBJECT_IDS } from 'src/engine/workspace-manager/workspace-sync-metadata/constants/standard-object-ids';
+import { WorkspaceIsSystem } from 'src/engine/twenty-orm/decorators/workspace-is-system.decorator';
+import { WorkspaceIsNotAuditLogged } from 'src/engine/twenty-orm/decorators/workspace-is-not-audit-logged.decorator';
+import { WorkspaceIsNullable } from 'src/engine/twenty-orm/decorators/workspace-is-nullable.decorator';
+import { WorkspaceIndex } from 'src/engine/twenty-orm/decorators/workspace-index.decorator';
+
+@WorkspaceEntity({
+  standardId: STANDARD_OBJECT_IDS.invitation,
+  namePlural: 'invitations',
+  labelSingular: 'Invitation',
+  labelPlural: 'Invitations',
+  description: 'An invitation',
+  icon: 'IconUserCircle',
+  labelIdentifierStandardId: INVITATION_STANDARD_FIELD_IDS.invitationToken,
+})
+@WorkspaceIsSystem()
+@WorkspaceIsNotAuditLogged()
+export class InvitationWorkspaceEntity extends BaseWorkspaceEntity {
```

Comment:
So there are two worlds that leave next to another in the codebase:
- The core/metadata schema, which are closer to traditional multi-tenant codebases, no-schema level isolation but a foreign key on each table, traditional TypeORM + optional nestjs-query package to generate resolvers
- The workspace-specific schemas. For that the logic is mostly homemade, we've got our own metadata definition language, our own 'TwentyORM', several abstractions layers to automatically generate the graphql/REST API.

It's not an easy task to distinguish what should go where. The most controversial objects today would be "Views" and "Workflows", two objects that belong to the workspace-schemas but could live outside as well.

The way I view it:
(1) does it hold PIIs? then workspace-schema
(2) should users be able to perform all CRUD operation or do we need a more fined-grained control? if specific resolvers/logic is triggered and we cannot allow all operations, then it probably shouldn't be in the workspace schema
(3) do we need to have list views / record show page generated automatically or is this mostly an internal object that will remain in settings? if we do need record pages, then workspace-schema

I'm curious what led you to switch half-way? It's possible I'm missing something
In the related issue I mentioned relying on `appTokens` (but I think the other directions doesn't make much sense, it's an old issue with unclear instructions). Instinctively I would also have kept the `sendInvites` mutation and also kept the storage for invitation email/token within the `core` schema. Mostly for reason (2), as I feel like it's really a 'system thing' and not a 'this is the workspace's business data' problem.

Open to the debate on this as this is subjective and I often changed my mind in the past on similar discussions.